controller/ProviderAdmin.php

```diff
@@ -31,6 +39,30 @@
  */
 class ProviderAdmin extends tao_actions_SaSModule
 {
+    public function addLtiInstanceForm()
+    {
+        $form = $this->getProviderFormFactory()->create();
+
+        if ($form->isSubmited() && $form->isValid()) {
+
+            $instance = $this->createInstance([RdfLtiProviderRepository::CLASS_URI], $form->getValues());
+
+            $this->setData('message', __('%s created', $instance->getLabel()));
+            $this->setData('reload', true);
+        }
+
+        $this->setData('formTitle', __('Edit Instance'));
+        $this->setData('myForm', $form->render());
+        $this->setView('form.tpl', 'tao');
+
+        return $form;
+    }
```

Comment:
IMO we should not duplicate code. Basically 90% here is duplicated. Instead, we can override the form fields by creating a black list of fields (fields to hide).

This can be done by DI in the original form (as @boajer did in his PR). I explained this to @bartlomiejmarszal. If it is not clear we can discuss.